trailblazer/services/analysis_service/utils.py

```diff
@@ -41,3 +43,24 @@ def create_update_analyses_response(analyses: list[Analysis]) -> UpdateAnalysesR
         analysis_data = analysis.to_dict()
         response_data.append(analysis_data)
     return UpdateAnalysesResponse(analyses=response_data)
+
+
+def get_upload_date(analysis: Analysis) -> datetime | None:
+    completed_job: Job | None = _get_completed_job(analysis.upload_jobs)
+    if analysis.workflow != Workflow.FASTQ or not completed_job:
+        return
+    return _get_completed_at_date(completed_job)
+
+
+def _get_completed_job(jobs: list[Job]) -> Job | None:
+    for job in jobs:
+        if _is_completed_job(job):
+            return job
+
+
+def _is_completed_job(job: Job):
+    return job.status == SlurmJobStatus.COMPLETED and job.started_at and job.elapsed
+
+
+def _get_completed_at_date(job: Job) -> datetime:
+    return job.started_at + timedelta(seconds=job.elapsed)
```

Comment:
Most upload jobs finish within minutes, so you could argue that this is irrelevant given that we just want the day that the job